From e1abb3b6dda3133a5348e824bddb235fc87eee16 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Thu, 27 Mar 2014 13:49:26 +0000
Subject: [PATCH 172/351] regulator: arizona-ldo1: Only process DT if no pdata

In arizona-core we only process the DT if no pdata is present, we should
be consistent in this approach. This patch adds similar handling into
this driver.

Change-Id: Ibde45a2d8d9151a477734b801fce6e8902360eeb
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/regulator/arizona-ldo1.c |   11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/regulator/arizona-ldo1.c b/drivers/regulator/arizona-ldo1.c
index bcd21ba..7f1660c 100644
--- a/drivers/regulator/arizona-ldo1.c
+++ b/drivers/regulator/arizona-ldo1.c
@@ -273,9 +273,11 @@ static int arizona_ldo1_probe(struct platform_device *pdev)
 	config.driver_data = ldo1;
 	config.regmap = arizona->regmap;
 
-	ret = arizona_ldo1_of_get_pdata(arizona, &config);
-	if (ret < 0)
-		return ret;
+	if (!dev_get_platdata(arizona->dev)) {
+		ret = arizona_ldo1_of_get_pdata(arizona, &config);
+		if (ret < 0)
+			return ret;
+	}
 
 	config.ena_gpio = arizona->pdata.ldoena;
 
@@ -299,7 +301,8 @@ static int arizona_ldo1_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	arizona_ldo1_of_put_pdata(&config);
+	if (!dev_get_platdata(arizona->dev))
+		arizona_ldo1_of_put_pdata(&config);
 
 	platform_set_drvdata(pdev, ldo1);
 
-- 
1.7.9.5

